.github/workflows/tests.yml

```
@@ -1,52 +1,58 @@
----
-
-name: Tests
+name: Tests with Tox
 
 on: [push, pull_request, workflow_dispatch]
 
 env:
   FORCE_COLOR: 1
 
+concurrency:
+  group: check-tests-tox-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

Also cancel repeat builds on `main`?
```suggestion
  cancel-in-progress: true
```

---

I would keep it running on main branch always, so how about cancel of not push to main?

---

No strong preference, this isn't a very busy repo so that sounds fine.